### PR TITLE
fix(github): Update CODEOWNERS for rnd directories

### PR DESCRIPTION
**Pull request by andrewhooker2**
### Background
We have added a new project with the directory `rnd/` containing several subdirectories. This PR updates the CODEOWNERS file to include maintainers for these new directories, ensuring proper code ownership and review processes.

### Changes 🏗️
- Updated the CODEOWNERS file in the `.github` directory to include new entries for the `rnd/` subdirectories.
- Added a new maintainer group `@Significant-Gravitas/rnd-maintainers` for the new project.
- Assigned the new maintainer group to all subdirectories of `rnd/` except for `infra/`.

### PR Quality Scorecard ✨
- [x] Have you used the PR description template? &ensp; `+2 pts`
- [x] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [x] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Updated the CODEOWNERS file to include new entries for the `rnd/` subdirectories.
- Added a new maintainer group `@Significant-Gravitas/rnd-maintainers`.
- Assigned the new maintainer group to all subdirectories of `rnd/` except for `infra/`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CODEOWNERS</strong><dd><code>Update CODEOWNERS for new `rnd/` subdirectories</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/CODEOWNERS

<li>Added new entries for <code>rnd/</code> subdirectories.<br> <li> Assigned <code>@Significant-Gravitas/rnd-maintainers</code> to new directories.<br>

</details>

  </td>
  <td><a href="https://github.com/Significant-Gravitas/AutoGPT/pull/7684/files#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by Torantulino** [comment]
> **Maintainability**	
> Consolidate multiple `rnd/` directory entries into a single wildcard entry
> **Consolidate the entries for the `rnd/` subdirectories under a single pattern to simplify the CODEOWNERS file and make future maintenance easier. This avoids redundancy and potential errors in future updates.**
> 
> [.github/CODEOWNERS [6-10]](https://github.com/Significant-Gravitas/AutoGPT/pull/7684/files#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7R6-R10)
> 
> ```diff
> -rnd/auto-gpt_builder/  @Significant-Gravitas/rnd-maintainers
> -rnd/autogpt_libs/      @Significant-Gravitas/rnd-maintainers
> -rnd/autogpt_server/    @Significant-Gravitas/rnd-maintainers
> -rnd/example_files/     @Significant-Gravitas/rnd-maintainers
> -rnd/market/            @Significant-Gravitas/rnd-maintainers
> +rnd/*  @Significant-Gravitas/rnd-maintainers
>  
> ```
> 
> * [ ]  **Apply this suggestion**
> 
> Suggestion importance[1-10]: 9
> Why: The suggestion to consolidate the entries for the `rnd/` subdirectories into a single wildcard entry improves maintainability by reducing redundancy and minimizing potential errors in future updates. The existing code is correctly identified, and the improved code accurately reflects the suggestion.
> 
> 9

Would this produce the same result?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -3,3 +3,8 @@ autogpt/   @Significant-Gravitas/maintainers
 forge/     @Significant-Gravitas/forge-maintainers
 benchmark/          @Significant-Gravitas/benchmark-maintainers
 frontend/           @Significant-Gravitas/frontend-maintainers
+rnd/auto-gpt_builder/  @Significant-Gravitas/rnd-maintainers
+rnd/autogpt_libs/      @Significant-Gravitas/rnd-maintainers
+rnd/autogpt_server/    @Significant-Gravitas/rnd-maintainers
+rnd/example_files/     @Significant-Gravitas/rnd-maintainers
+rnd/market/            @Significant-Gravitas/rnd-maintainers
```
Comment (ntindle) [inline .github/CODEOWNERS]: Love the idea, use maintainers without rnd- though

**Review by ntindle — CHANGES_REQUESTED** [review]

**Comment by ntindle** [comment]
> > **Maintainability**
> > Consolidate multiple `rnd/` directory entries into a single wildcard entry
> > **Consolidate the entries for the `rnd/` subdirectories under a single pattern to simplify the CODEOWNERS file and make future maintenance easier. This avoids redundancy and potential errors in future updates.**
> > [.github/CODEOWNERS [6-10]](https://github.com/Significant-Gravitas/AutoGPT/pull/7684/files#diff-3d36a1bf06148bc6ba1ce2ed3d19de32ea708d955fed212c0d27c536f0bd4da7R6-R10)
> > ```diff
> > -rnd/auto-gpt_builder/  @Significant-Gravitas/rnd-maintainers
> > -rnd/autogpt_libs/      @Significant-Gravitas/rnd-maintainers
> > -rnd/autogpt_server/    @Significant-Gravitas/rnd-maintainers
> > -rnd/example_files/     @Significant-Gravitas/rnd-maintainers
> > -rnd/market/            @Significant-Gravitas/rnd-maintainers
> > +rnd/*  @Significant-Gravitas/rnd-maintainers
> >  
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Suggestion importance[1-10]: 9
> > Why: The suggestion to consolidate the entries for the `rnd/` subdirectories into a single wildcard entry improves maintainability by reducing redundancy and minimizing potential errors in future updates. The existing code is correctly identified, and the improved code accurately reflects the suggestion.
> 
> Would this produce the same result?

We also have infra in there that should be under devops probs

**Comment by ntindle** [comment]
@Torantulino made me think about it and they run top to bottom so defining a root * one works and then overwriting with infra

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,5 +1,6 @@
-.github/workflows/  @Significant-Gravitas/devops
-autogpt/   @Significant-Gravitas/maintainers
-forge/     @Significant-Gravitas/forge-maintainers
-benchmark/          @Significant-Gravitas/benchmark-maintainers
-frontend/           @Significant-Gravitas/frontend-maintainers
+* @Significant-Gravitas/maintainers
+.github/workflows/ @Significant-Gravitas/devops
+forge/ @Significant-Gravitas/forge-maintainers
+benchmark/ @Significant-Gravitas/benchmark-maintainers
+frontend/ @Significant-Gravitas/frontend-maintainers
```
Comment (ntindle) [inline .github/CODEOWNERS:5]: this is an existing bug, they don't have write access to the repo

**Review by ntindle — APPROVED** [review]

**Review by ntindle — APPROVED** [review]

**Comment by ntindle** [comment]
Fixed #7683

**Pull request merged**